gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreUtils.java

```diff
@@ -421,6 +425,19 @@ private static Deserializer getDeserializer(HiveRegistrationUnit unit) {
     return deserializer;
   }
 
+  public static void updateColumnsInfo(HiveSpec spec) throws IOException {
+    HiveTable table = spec.getTable();
+    if (table.getSerDeProps().contains(serdeConstants.LIST_COLUMNS)) {
+      if (table.getSerDeManager().isPresent()) {
+        String path = spec.getPartition().isPresent() && spec.getPartition().get().getLocation().isPresent() ? spec.getPartition()
```

Comment:
just for my understanding: is this just a optimization to use partition level path ?

Reply:
Yeah, in OrcSerDeManager, if no avro shcema literal is set, it will try to list the path to get the latest file's schema, even it's not gonna to happen in our case, just want to make sure the list operation is as small as possible even something goes wrong